cmd/storage-class.go

```diff
@@ -0,0 +1,213 @@
+	if rrsParity < minimumParityDisks {
+		return errors.New("Reduced redundancy storage class parity should be greater than or equal to " + strconv.Itoa(minimumParityDisks))
+	}
+
+	// Reduced redundancy implies lesser parity than standard storage class. So, RRS parity disks should be
+	// - less than N/2, if StorageClass parity is not set.
+	// - less than StorageClass Parity, if Storage class parity is set.
+	switch ssParity {
+	case 0:
+		if rrsParity >= len(globalEndpoints)/2 {
+			return errors.New("Reduced redundancy storage class parity disks should be less than " + strconv.Itoa(len(globalEndpoints)/2))
+		}
+	default:
+		if rrsParity >= ssParity {
+			return errors.New("Reduced redundancy storage class parity disks should be less than " + strconv.Itoa(ssParity))
+		}
+	}
+
+	return nil
+}
+
+// Validates the parity disks for Standard storage class
+func validateSSParity(ssParity, rrsParity int) (err error) {
+
+	// Standard storage class implies more parity than Reduced redundancy storage class. So, Standard storage parity disks should be
+	// - greater than or equal to 2, if RRS parity is not set.
+	// - greater than RRS Parity, if RRS parity is set.
+	switch rrsParity {
+	case 0:
+		if ssParity < minimumParityDisks {
+			return errors.New("Standard storage class parity disks should be greater than or equal to " + strconv.Itoa(minimumParityDisks))
+		}
+	default:
+		if ssParity <= rrsParity {
+			return errors.New("Standard storage class parity disks should be greater than " + strconv.Itoa(rrsParity))
+		}
+	}
+
+	// Standard storage class parity should be less than or equal to N/2
+	if ssParity > len(globalEndpoints)/2 {
+		return errors.New("Standard storage class parity disks should be less than or equal to " + strconv.Itoa(len(globalEndpoints)/2))
+	}
+
+	return nil
+}
+
+// Returns the data and parity drive count based on storage class
+// If storage class is set using the env vars MINIO_STORAGE_CLASS_RRS and MINIO_STORAGE_CLASS_STANDARD
+// -- corresponding values are returned
+// If storage class is not set using environment variables, default values are returned
+// -- Default for Reduced Redundancy Storage class is, parity = 2 and data = N-Parity
+// -- Default for Standard Storage class is, parity = N/2, data = N/2
+// If storage class is not present in metadata, default value is data = N/2, parity = N/2
+func getDrivesCount(sc string, disks []StorageAPI) (data, parity int) {
+	totalDisks := len(disks)
+	switch sc {
+	case reducedRedundancyStorageClass:
+		if globalRRStorageClass.Parity != 0 {
+			// set the rrs parity if available
+			parity = globalRRStorageClass.Parity
+		} else {
+			// else fall back to default value
+			parity = defaultRRSParity
+		}
+	case standardStorageClass:
+		if globalStandardStorageClass.Parity != 0 {
+			// set the standard parity if available
+			parity = globalStandardStorageClass.Parity
+		} else {
+			// else fall back to default value
+			parity = totalDisks / 2
```

Comment:
You could initialize parity around line 146 to `totalDisks / 2`, and then remove default cases in lines 162 and 165